plugin/build.gradle.kts

```diff
@@ -82,6 +129,27 @@ gradlePlugin {
     }
 }
 
+// Need to move publishing configuration into afterEvaluate {}
+// to override changes done by "com.gradle.plugin-publish" plugin in afterEvaluate {} block
+// See PublishPlugin class for details
+afterEvaluate {
+    publishing {
+        publications {
+            withType(MavenPublication::class.java) {
+                // Special workaround to publish shadow jar instead of normal one. Name to override peeked here:
+                // https://github.com/gradle/gradle/blob/master/subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/MavenPluginPublishPlugin.java#L73
+                if (name == "pluginMaven") {
+                    setArtifacts(
+                        listOf(
+                            project.tasks.named("shadowJar").get()
```

Comment:
Don't you already have the `shadowJarTask` in scope?

Reply:
Good point, will use it :+1:

Reply:
Updated